.github/CONTRIBUTOR_GUIDE/ct/AppName.md

```diff
@@ -0,0 +1,231 @@
+- Befoer updating, check if a new version exists. 
+  - We use the `${APPLICATION}_version.txt` file created in `/opt` during the install to compare new versions against the currently installed version.
+
+Example with a Github Release:
+```bash
+ RELEASE=$(curl -fsSL https://api.github.com/repos/snipe/snipe-it/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
+  if [[ ! -f /opt/${APP}_version.txt ]] || [[ "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]]; then
+    msg_info "Updating ${APP} to v${RELEASE}"
+    #DO UPDATE
+  else
+    msg_ok "No update required. ${APP} is already at v${RELEASE}."
+  fi
+  exit
+}
+```
+### 3.4 **Verbosity**
+- Use the appropriate flag (**-q** in the examples) for a command to suppress its output.
+Example:
+```bash
+wget -q
+unzip -q
+```
+- If a command does not come with this functionality use `&>/dev/null` to suppress it's output.
+
+Example:
+```bash
+php artisan migrate --force &>/dev/null
+php artisan config:clear &>/dev/null
+```
+
+### 3.5 **Backups**
+- Backup user data if necessary.
+- Move all user data back in the directory when the update is finished.
+>[!NOTE]
+>This is not meant to be a permanent backup
+
+Example backup:
+```bash
+  mv /opt/snipe-it /opt/snipe-it-backup
+```
+Example config restore:
+```bash
+  cp /opt/snipe-it-backup/.env /opt/snipe-it/.env
+  cp -r /opt/snipe-it-backup/public/uploads/ /opt/snipe-it/public/uploads/
+  cp -r /opt/snipe-it-backup/storage/private_uploads /opt/snipe-it/storage/private_uploads
+```
+
+### 3.6 **Cleanup**
+- Do not forget to remove any temporary files/folders such as zip-files or temporary backups.
+Example:
+```bash
+  rm -rf /opt/v${RELEASE}.zip
+  rm -rf /opt/snipe-it-backup
+```
+
+### 3.7 **No update function**
+- In case you can not provide a update function use the following code to provide user feedback.
+```bash
+function update_script() {
+    header_info
+    check_container_storage
+    check_container_resources
+    if [[ ! -d /opt/snipeit ]]; then
+        msg_error "No ${APP} Installation Found!"
+        exit
+    fi
+    msg_error "Ther is currently no automatic update function for ${APP}."
+    exit
+}
+```
+
+---
+
+## 4 **End of the script**
+- `start`: Launches Whiptail dialogue
+- `build_container`: Collects and integrates user settings
+- `description`: Sets LXC container description
+- With `echo -e "${TAB}${GATEWAY}${BGN}http://${IP}${CL}"` you can point the user to the IP:PORT/folder needed to access the app.
+
+```bash
+start
+build_container
+description
+
+msg_ok "Completed Successfully!\n"
+echo -e "${CREATING}${GN}${APP} setup has been successfully initialized!${CL}"
+echo -e "${INFO}${YW} Access it using the following URL:${CL}"
+echo -e "${TAB}${GATEWAY}${BGN}http://${IP}${CL}"
+```
+---
+
+## 5. **Contribution checklist**
+
+- [ ] Shebang is correctly set (`#!/usr/bin/env bash`).
+- [ ] Correct link to *build.func*
+- [ ] Metadata (author, license) is included at the top.
```

Comment:
I think this chapter can be removed. This file is more a "rundown of an install script".

This checklist may be helpful in the PR template.

Reply:
I think it is better placed here. When you finished your script and are at the bottom of the guide you can check yourself. This would make the PR template to convoluted i think.